library/src/main/java/com/google/android/exoplayer/extractor/flv/AudioTagPayloadReader.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2014 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.exoplayer.extractor.flv;
+
+import android.util.Log;
+import android.util.Pair;
+
+import com.google.android.exoplayer.C;
+import com.google.android.exoplayer.MediaFormat;
+import com.google.android.exoplayer.extractor.TrackOutput;
+import com.google.android.exoplayer.util.CodecSpecificDataUtil;
+import com.google.android.exoplayer.util.MimeTypes;
+import com.google.android.exoplayer.util.ParsableBitArray;
+import com.google.android.exoplayer.util.ParsableByteArray;
+
+import java.util.Collections;
+
+/**
+ * Parses audio tags of from an FLV stream and extracts AAC frames.
+ */
+final class AudioTagPayloadReader extends TagPayloadReader {
+
+  private static final String TAG = "AudioTagPayloadReader";
+
+  // Sound format
+  private static final int AUDIO_FORMAT_LINEAR_PCM_PLATFORM_ENDIAN = 0;
+  private static final int AUDIO_FORMAT_ADPCM = 1;
+  private static final int AUDIO_FORMAT_MP3 = 2;
+  private static final int AUDIO_FORMAT_LINEAR_PCM_LITTLE_ENDIAN = 3;
+  private static final int AUDIO_FORMAT_NELLYMOSER_16KHZ_MONO = 4;
+  private static final int AUDIO_FORMAT_NELLYMOSER_8KHZ_MONO = 5;
+  private static final int AUDIO_FORMAT_NELLYMOSER = 6;
+  private static final int AUDIO_FORMAT_G711_A_LAW = 7;
+  private static final int AUDIO_FORMAT_G711_MU_LAW = 8;
+  private static final int AUDIO_FORMAT_RESERVED = 9;
+  private static final int AUDIO_FORMAT_AAC = 10;
+  private static final int AUDIO_FORMAT_SPEEX = 11;
+  private static final int AUDIO_FORMAT_MP3_8KHZ = 14;
+  private static final int AUDIO_FORMAT_DEVICE_SPECIFIC = 15;
+
+  // AAC PACKET TYPE
+  private static final int AAC_PACKET_TYPE_SEQUENCE_HEADER = 0;
+  private static final int AAC_PACKET_TYPE_AAC_RAW = 1;
+
+  // SAMPLING RATES
+  private static final int[] AUDIO_SAMPLING_RATE_TABLE = new int[] {
+      5500, 11000, 22000, 44000
+  };
+
+  // Variables to hold audio configuration.
+  private int format;
+  private int sampleRate;
```

Comment:
Unused (ditto x2 below). format would also be used if you merge shouldParsePayload into parseHeader.